x-pack/plugins/index_management/public/application/components/mappings_editor/lib/mappings_validator.ts

```diff
@@ -277,32 +285,54 @@ export const validateMappingsConfiguration = (
   return { value: copyOfMappingsConfig, errors };
 };
 
-export const validateMappings = (mappings: any = {}): MappingsValidatorResponse => {
+const validatePluginsParameters = (
+  mappingsConfiguration: GenericObject,
+  esNodesPlugins: string[]
+): { value: any; errors: MappingsValidationError[] } => {
+  const copyOfMappingsConfig = { ...mappingsConfiguration };
+  const errors: MappingsValidationError[] = [];
+
+  // Mapper size plugin parameters
+  if ('_size' in copyOfMappingsConfig && !esNodesPlugins.includes(MapperSizePluginId)) {
+    errors.push({
+      code: 'ERR_CONFIG',
+      configName: '_size',
+    });
+    delete copyOfMappingsConfig._size;
+  }
+
+  return { value: copyOfMappingsConfig, errors };
+};
+
+export const validateMappings = (
+  mappings: unknown = {},
+  esNodesPlugins: string[] = []
+): MappingsValidatorResponse => {
   if (!isPlainObject(mappings)) {
     return { value: {} };
   }
 
-  const { properties, dynamic_templates: dynamicTemplates, ...mappingsConfiguration } = mappings;
+  const {
+    properties,
+    dynamic_templates: dynamicTemplates,
+    ...mappingsConfiguration
+  } = mappings as GenericObject;
 
-  const { value: parsedConfiguration, errors: configurationErrors } =
+  const { value: parsedConfiguration1, errors: configurationErrors } =
     validateMappingsConfiguration(mappingsConfiguration);
+  const { value: parsedConfiguration2, errors: pluginsErrors } = validatePluginsParameters(
```

Comment:
nit: could we rename this to `parsedConfiguration2`=>`parsedParameters` and `parsedConfiguration1`=>`parsedConfiguration`

Reply:
True that it was not optimal. I refactored it in https://github.com/elastic/kibana/pull/119365/commits/87f54be56a69a0c859094808aa784e7942cf0ac9